test/test_torch.py

```diff
@@ -6266,39 +6266,32 @@ def test_bitwise_xor(self, device):
     def test_logical_not(self, device, dtype):
         data = [10, 1, 0.3, 0, -0.3, -1, -10]
         a = torch.tensor(data, dtype=dtype, device=device)
-
-        # do this before constructing the numpy array because np can't construct
-        # bfloat16 tensors.  Can we define our own dtype in NumPy so testing would be easier?
-        if dtype == torch.bfloat16 or dtype.is_complex:
-            self.assertRaises(RuntimeError, lambda: a.logical_not())
-            self.assertRaises(RuntimeError, lambda: a.logical_not_())
-            raise unittest.SkipTest('logical_not not supported on {}'.format(dtype))
-
-        a_np = np.array(data, dtype=torch_to_numpy_dtype_dict[dtype])
-        self.assertEqual(np.logical_not(a_np), torch.logical_not(a).to('cpu'))
-        self.assertEqual(np.logical_not(a_np, out=a_np), a.logical_not_().to('cpu'))
+        if dtype == torch.bfloat16 or dtype.is_complex:  # numpy doesn't support these dtypes
```

Comment:
NumPy does support logical not on complex types, doesn't it? 

```
a = np.array((1 + 1j))
np.logical_not(a)
: False
```

Reply:
For bfloat16 you can set an explicit dtype of float32 with a pattern like:

```
if dtype is torch.bfloat16:
  np_dtype = np.float32
else:
  np_dtype = torch_to_numpy_dtype_dict[dtype]
```

Reply:
Sorry for the late reply. I've removed complex from this line.

I've tried your suggestion but `self.assertEqual(a, a.cpu().numpy())` (a few lines below) always leads to `TypeError` for bfloat16.

I have also rebased the stack.